### DISCUSSION TITLE
Int16 against

### DISCUSSION THREAD

**Discussion by pendurkov** [benclaussen/NetboxPS#48] [Q&A]
I have environment with devices with over 300 000 ids and when i add new machine i got that. 

PS C:\Windows\system32> Set-NetboxVirtualMachine -Primary_IP4 $NBip.id -Id $vm.id
Set-NetboxVirtualMachine : Cannot process argument transformation on parameter 'Primary_IP4'. Cannot convert value
"334039" to type "System.UInt16". Error: "Value was either too large or too small for a UInt16."

Could you please, advice what i am missing

**Comment by benclaussen1** [comment]
I apologize. In my early development and learning, this was a limitation I unintentionally added long ago by setting the ID parameter type to `uint16`. You can manually edit the `Netbox.psm1` file and update the `Set-NetboxVirtualMachine` function `$Id` parameter type to `[int64]` to workaround this for now until I can push out a new release. Between other work and life commitments I have not been able to focus on this in some time.

**Comment by benclaussen1** [comment]
Moved to issue #49

**Comment by pendurkov** [comment]
Thanks for reply and all work, Ben. I tried to change but probably it`s a big deeper. When i run the standard function New-NBVirtualMachine - VM is created into Netbox, eth0 is created, IP is created too, but unable to assign the IP to eth0 . I even tried to change all UInt16 to int64 into psm1 but still got that.

Set-NetboxIPAMAddress : Cannot process argument transformation on parameter 'Assigned_Object_Id'. Cannot convert value
"95171" to type "System.UInt16". Error: "Value was either too large or too small for a UInt16."
At line:27 char:100
+ ...  virtualization.vminterface -Assigned_Object_Id $interface.id -id $NB ...
+                                                     ~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Set-NetboxIPAMAddress], ParameterBindingArgumentTransformationExceptio
   n
    + FullyQualifiedErrorId : ParameterArgumentTransformationError,Set-NetboxIPAMAddress

Set-NetboxVirtualMachine : Cannot process argument transformation on parameter 'Primary_IP4'. Cannot convert value
"334089" to type "System.UInt16". Error: "Value was either too large or too small for a UInt16."
At line:28 char:47
+         Set-NetboxVirtualMachine -Primary_IP4 $NBip.id -Id $vm.id
+                                               ~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Set-NetboxVirtualMachine], ParameterBindingArgumentTransformationExcep
   tion
    + FullyQualifiedErrorId : ParameterArgumentTransformationError,Set-NetboxVirtualMachine

**Reply by alagoutte** [reply]
do you have to reimport module with force ? import-module ... -force

**Reply by pendurkov** [reply]
HI Alex, I usually edit the psm1, then save and close all powershell windows and do import-module netboxps etc. Now i tried with -force parameter but still no luck.
